pages/map.py

```diff
@@ -324,13 +326,19 @@ def process_trips_group(trips_group):
 
 @callback(
     Output('store-trips-map', 'data'),
+    Output('no-trip-text', 'style'),
+    Output('trip-map', 'style'),
     Input('store-trips', 'data'),
 )
 def store_trips_map_data(trips_data):
+    # if there is no trips_data for selected date, show 'no-trip-text'
+    if trips_data == None or 'length' not in trips_data or trips_data['length'] == 0:
```

Comment:
This can be simplified:
```suggestion
    if not trips_data:
```

Empty lists and dictionaries are falsy in Python, so whether it is `{}` `[]` or `None`, this expression treats it the same.

By the way if you did need to specifically check the length, the usual way to do it is `len(trips_data)`